cow_instrument/InstrumentTree.cpp

```diff
@@ -97,15 +100,20 @@ InstrumentTree InstrumentTree::modify(size_t node,
                       currentChildren.end());
     }
   }
-  return InstrumentTree(std::move(newNodes), m_detectorVec.size());
+  return std::unique_ptr<const InstrumentTree>(
+      new InstrumentTree(std::move(newNodes), m_detectorVec.size()));
 }
 
-InstrumentTree InstrumentTree::modify(const Node *node,
-                                      const Command &command) const {
+std::unique_ptr<const InstrumentTree>
```

Comment:
@OwenArnold My point is: Why should we force the client to do this? If we create a type for this it can be handled internally, and be having a `cow_ptr` we get rid of the issue about potentially adding non-const methods. So, similar to how `BinEdges` contains a `cow_ptr<HistogramX>`, we could have a `Instrument` that contains a `cow_ptr` to the underlying data.